### PR TITLE
Batch index handler

### PR DESCRIPTION
**Pull request by truhacevkir**
- The handler reads keys from persisted resources bucket
- Keys are persisted in second bucket in batches of 10

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -629,3 +648,14 @@ Resources:
...
+  KeyBatchesBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      AccessControl: Private
+      BucketName: !Sub "${KeyBatchesBucketName}-${AWS::AccountId}"
+      LifecycleConfiguration:
+        Rules:
+          - Id: DeleteContentAfter3Days
+            Status: Enabled
+            ExpirationInDays: 3
```
Comment (anetteOlli) [inline template.yaml:661]: NB! make sure this is not too short span. We have had issues with this in cristin-import.

File: template.yaml [hunk 2]
```diff
@@ -582,13 +594,20 @@ Resources:
           BATCH_INDEX_EVENT_BUS_NAME: !GetAtt BatchIndexEventBus.Name
           PERSISTED_RESOURCES_PATH: !Sub "s3://${ResourcesBucket}/resources"
 
-  BatchIndexEventBus:
-    Type: AWS::Events::EventBus
+  GenerateKeyBatchesHandler:
```
Comment (anetteOlli) [inline template.yaml:597]: Can this name indicate that it is for the search-index?

File: template.yaml [hunk 3]
```diff
@@ -463,6 +467,14 @@ Resources:
                 responsePayload:
                   topic: [ "PublicationService.ExpandedEntry.Persisted" ]
 
+  BatchIndexEventBus:
```
Comment (anetteOlli) [inline template.yaml:470]: Note that max age for events on aws events bus is 6 hours

**Review by anetteOlli — DISMISSED** [review]

**Review by anetteOlli — APPROVED** [review]

File: batch-index/src/test/java/no/unit/nva/search/GenerateKeyBatchesHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+        var actual = getPersistedFileFromOutputBucket();
+        var expectedNumberOfFiles = getExpectedNumberOfFiles(numberOfItemsInBucket);
+
+        assertThat(actual.size(), is(equalTo(expectedNumberOfFiles)));
+        assertThat(actual.stream().collect(Collectors.joining(System.lineSeparator())), is(equalTo(expected)));
+    }
+
+    private static int getExpectedNumberOfFiles(int numberOfItemsInBucket) {
+        return (int) Math.ceil(numberOfItemsInBucket < 11 ? 1 : (double) numberOfItemsInBucket / 10);
```
Comment (sondrev) [inline batch-index/src/test/java/no/unit/nva/search/GenerateKeyBatchesHandlerTest.java:59]: 10 should be a constant

This method was hard to follow. Maybe split the edgecase of < 11 outside of the Math.ceil?
